core/src/main/java/org/apache/accumulo/core/client/ClientConfiguration.java

```diff
@@ -50,7 +50,7 @@
  * @deprecated since 2.0.0, replaced by {@link Accumulo#newClient()}
  */
 @Deprecated
-public class ClientConfiguration {
+public class ClientConfiguration extends CompositeConfiguration {
```

Comment:
This went through a deprecation cycle in 1.9. It's not needed to re-add this. We intentionally removed this to avoid non-public API leakage of commons-configuration stuff.

Reply:
Why remove the inheritance?  If we are keeping the class for backwards compatibility, removing the super class may still break existing code that uses it. 

Also looks like this class wasn't deprecated in 1.9: https://github.com/apache/accumulo/blob/1.9/core/src/main/java/org/apache/accumulo/core/client/ClientConfiguration.java#L61

Reply:
Yes, the whole point was to break code that used it in 2.0. This is why we bumped versions to 1.9 instead of staying on 1.8... for the sole purpose of deprecating out this superclass. This was documented in the release notes for 1.9.0.

https://accumulo.apache.org/release/accumulo-1.9.0/#deprecated-clientconfiguration-api-using-commons-config

As for marking it deprecated, Java has no way of marking the superclass as deprecated. So, what we did instead was deprecate all of the methods inherited from the superclass:

https://github.com/apache/accumulo/blob/1.9/core/src/main/java/org/apache/accumulo/core/client/ClientConfiguration.java#L140

It was only during the 2.0 development that we decided to deprecate the class in its entirety, in favor of using properties... but that's a separate issue. The methods inherited from the superclass were declared as deprecated in 1.9... all of them, and we documented this in the release notes.

We should *not* re-add this API leakage back into the API after already removing it, when we've already done as much as we have to prepare users, and it fixes a legitimate API leakage.

Reply:
IIRC this type is from commons config 1 and causes problems when trying to use commons config 2?

Reply:
Yes, the type is incompatible with commons-configuration2. However, that problem exists regardless of whether it is in the public API or not. But, having it in the public API prevents us from migrating to commons-configuration2. So, it needs to be out of the public API so we can update it, whether to match what Hadoop is using, for class path convergence, or for security updates. In all such cases, we cannot upgrade it if it remains in the public API.

Reply:
Gotcha.  I'll drop this fix.

Reply:
> But, having it in the public API prevents us from migrating to commons-configuration2.

Also causes problems for users or dependencies using commons config 2